src/Tokenizer/FCT.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer;
+
+// Forward Compatibility Tokens
+
+if (\PHP_VERSION_ID >= 8_04_00) {
+    final class FCT
+    {
+        public const T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG = T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG;
+        public const T_ATTRIBUTE = T_ATTRIBUTE;
+        public const T_ENUM = T_ENUM;
+        public const T_MATCH = T_MATCH;
+        public const T_NULLSAFE_OBJECT_OPERATOR = T_NULLSAFE_OBJECT_OPERATOR;
+        public const T_PUBLIC_SET = T_PUBLIC_SET;
+        public const T_PROTECTED_SET = T_PROTECTED_SET;
+        public const T_PRIVATE_SET = T_PRIVATE_SET;
+        public const T_READONLY = T_READONLY;
+    }
+} elseif (\PHP_VERSION_ID >= 8_01_00) {
+    final class FCT
+    {
+        public const T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG = T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG;
+        public const T_ATTRIBUTE = T_ATTRIBUTE;
+        public const T_ENUM = T_ENUM;
+        public const T_MATCH = T_MATCH;
+        public const T_NULLSAFE_OBJECT_OPERATOR = T_NULLSAFE_OBJECT_OPERATOR;
+        public const T_PUBLIC_SET = -1;
+        public const T_PROTECTED_SET = -1;
+        public const T_PRIVATE_SET = -1;
+        public const T_READONLY = T_READONLY;
+    }
+} elseif (\PHP_VERSION_ID >= 8_00_00) {
+    final class FCT
+    {
+        public const T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG = -1;
+        public const T_ATTRIBUTE = T_ATTRIBUTE;
+        public const T_ENUM = -1;
+        public const T_MATCH = T_MATCH;
+        public const T_NULLSAFE_OBJECT_OPERATOR = T_NULLSAFE_OBJECT_OPERATOR;
+        public const T_PUBLIC_SET = -1;
+        public const T_PROTECTED_SET = -1;
+        public const T_PRIVATE_SET = -1;
+        public const T_READONLY = -1;
+    }
+} else {
+    final class FCT
+    {
+        public const T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG = -1;
+        public const T_ATTRIBUTE = -1;
+        public const T_ENUM = -1;
+        public const T_MATCH = -1;
+        public const T_NULLSAFE_OBJECT_OPERATOR = -1;
+        public const T_PUBLIC_SET = -1;
+        public const T_PROTECTED_SET = -1;
+        public const T_PRIVATE_SET = -1;
+        public const T_READONLY = -1;
```

Comment:
i have a feeling that somewhere we will see `T_READY === T_PRIVATE_SET`. let's have unique values for them.

Reply:
The idea I have is that:
- if the token is present, then the value of class constant is the value of the constant of its name
- if the token is not present, then the value of class constant is `-1`
- 
which is checked in `FCTTest`. Wouldn't that be enough? If not, let me know how to improve tests.

> `T_READY === T_PRIVATE_SET`

Do you mean comparing one constant to another? I believe we don't do it anywhere. If someone tries, PHPStan would complain about it as it would always be equal/different and I hope we would catch it in code review.

Reply:
> let's have unique values for them.

There is no harm in making them unique, done.

Reply:
```
$tokens = Tokens::fromArray([ ... manually prepared array, some items are FCT::... ]);
manipulateTokens($tokens);
$tokens[$index]->equals(FCT::T_READY); // can cause problems if all FCT are -1
```